src/analyses/useAfterFree.ml

```diff
@@ -0,0 +1,153 @@
+(** An analysis for the detection of use-after-free vulnerabilities. *)
+
+open GoblintCil
+open Analyses
+open MessageCategory
+
+module ToppedVarInfoSet = SetDomain.ToppedSet(CilType.Varinfo)(struct let topname = "All Heap Variables" end)
+
+module Spec : Analyses.MCPSpec =
+struct
+  include Analyses.DefaultSpec
+
+  let name () = "useAfterFree"
+
+  module D = ToppedVarInfoSet
+  module C = Lattice.Unit
+
+  (** TODO: Try out later in benchmarks to see how we perform with and without context-sensititivty *)
+  let context _ _ = ()
+
+
+  (* HELPER FUNCTIONS *)
+
+  let rec warn_lval_might_contain_freed ?(is_double_free = false) (transfer_fn_name:string) (lval:lval) ctx =
```

Comment:
Would it make sense to switch the order of  `ctx` and `lval` here? That might make invocations less awkward e.g. in `return`.

Reply:
I'll do the same for `warn_exp_might_contain_freed` for consistency's sake, as it's essentially dual to `warn_lval_might_contain_freed`

Reply:
Also done for `warn_exp_might_contain_freed` now (cf. https://github.com/goblint/analyzer/pull/1050/commits/6bfcf02a0e4bbda27846383265a92d7da3c4a062)